internal/go_repository.bzl

```diff
@@ -310,6 +324,24 @@ def _go_repository_impl(ctx):
     # Apply patches if necessary.
     patch(ctx)
 
+    if ctx.os.name.startswith("mac"):
+        arch = getattr(ctx.os, "arch", "x86")
+        if arch.startswith("x"):
+            buildozer = ctx.path(Label("@buildozer_macos_x86//file:buildozer"))
+        else:
+            buildozer = ctx.path(Label("@buildozer_macos_arm//file:buildozer"))
+    elif ctx.os.name.startswith("linux"):
+        buildozer = ctx.path(Label("@buildozer_linux_x86//file:buildozer"))
+    else:
+        buildozer = None
+        print("Unable to find buildozer binary, so omitting version information from tags")
+
+    if buildozer:
+        for tag in tags:
+            res = ctx.execute([buildozer, "add tags %s" % tag, "//...:%go_library"])
+            if res.return_code != 0:
+                fail("Unable to execute buildozer for tag %s" % tag)
+
```

Comment:
Aside from the earlier point I made about favoring usage of `gazelle` over `buildozer`, I think we should avoid modifying things after `patch(ctx)`. This way, user are always left with an option of overriding the default generated result with a custom patch. If we do these _after_ `patch(ctx)`, user will not have much options but to accept the default (or, patch Gazelle themselves).

Reply:
But we're not sure if there are build files being modified by the `patch`. If there are, then the result of running `buildozer` may not allow patches to apply cleanly. The way things are approached here feels like the least risky, and we should only be adding metadata at this point.

Reply:
FWIW... what we will be doing with rules_licenese is to rewrite the download after patch. We need to be able to rewrite license() rules based on scanning the code after download, and potentially updating a patched BUILD file.